include/bluetooth/services/bms_srv.h

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright (c) 2020 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#ifndef BT_GATT_BMS_SRV_H_
+#define BT_GATT_BMS_SRV_H_
+
+/**@file
+ * @defgroup bt_gatt_bms_srv Bond Management Service API
+ * @{
+ * @brief API for the Bond Management Service (BMS).
+ */
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include <stdlib.h>
+#include <zephyr/types.h>
+
+/** @brief BMS server callback structure. */
+struct bt_gatt_bms_srv_cb {
+	/** Operation authorization callback
+	 *
+	 *  If this callback is set, support for BMS Authorization Code
+	 *  functionality is enabled. This function is called whenever
+	 *  an authorized operation is requested by the client.
+	 *
+	 *  @param auth_code     Authorization Code
+	 *  @param auth_code_len Length of Authorization Code
+	 *
+	 * @retval true   If the authorization was successful.
+	 * @retval false  Otherwise.
+	 */
+	bool (*operation_authorize)(const uint8_t *auth_code,
```

Comment:
yes, I can shorten it.

I don't think there is a need for asynchronous authorization for now. We can add it later on if someone requests it. 

Additionally, you need to respond with "Insufficient Authorization" ATT error code in the Control Point write context if authorization codes do not match. So, Zephyr API could block adding this async functionality as well.

Reply:
Actually, the BLE ATT protocol requirements block this asynchronous functionality, since you need to respond immediately to the ATT Write Request. BMS would need to be designed differently (using indication instead of writable control point) to support asynchronous authorization.